packages/devtools_app/lib/src/shared/diagnostics/diagnostics_node.dart

```diff
@@ -591,10 +591,12 @@ class RemoteDiagnosticsNode extends DiagnosticableTree {
   bool get inHideableGroup {
     if (_alwaysVisible(this)) return false;
     final parentIsHideable = parent != null && !_alwaysVisible(parent!);
-    final childIsHideable =
+    final firstChildIsHideable =
         childrenNow.isNotEmpty && !_alwaysVisible(childrenNow.first);
 
-    return hasChildren ? parentIsHideable : parentIsHideable || childIsHideable;
+    return hasChildren
+        ? parentIsHideable || firstChildIsHideable
+        : parentIsHideable;
```

Comment:
nit: since `parentIsHideable` is in both the if and else cases, this could be `parentIsHideable || (hasChildren && firstChildIsHideable)`. If `hasChildren` and `childrenNow.isNotEmpty` are equivalent, this could technically just be `parentIsHideable || firstChildIsHideable`

Reply:
Done, thanks!